controllers/skipjob/controller.go

```diff
@@ -0,0 +1,133 @@
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// +kubebuilder:rbac:groups=skiperator.kartverket.no,resources=skipjobs;skipjobs/status,verbs=get;list;watch;update
+// +kubebuilder:rbac:groups=batch,resources=jobs;cronjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods;pods/ephemeralcontainers,verbs=get;list;watch;create;update;patch;delete
+
+type SKIPJobReconciler struct {
+	util.ReconcilerBase
+	RESTClient rest.Interface
+}
+
+func (r *SKIPJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		// GenerationChangedPredicate is now only applied to the SkipJob itself to allow status changes on Jobs/CronJobs to affect reconcile loops
+		For(&skiperatorv1alpha1.SKIPJob{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&batchv1.CronJob{}).
+		Owns(&batchv1.Job{}).
+		Watches(&batchv1.Job{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+			job, isJob := object.(*batchv1.Job)
+
+			if !isJob {
+				return nil
+			}
+
+			if job.Status.CompletionTime != nil {
+				return []reconcile.Request{
+					{
+						types.NamespacedName{
+							Namespace: job.Namespace,
+							Name:      job.Labels[SKIPJobReferenceLabelKey],
+						},
+					},
+				}
+			}
+
+			return nil
+		})).
+		Owns(&networkingv1.NetworkPolicy{}).
+		Owns(&istionetworkingv1beta1.ServiceEntry{}).
+		// Some NetPol entries are not added unless an application is present. If we reconcile all jobs when there has been changes to NetPols, we can assume
+		// that changes to an Applications AccessPolicy will cause a reconciliation of Jobs
+		Watches(&networkingv1.NetworkPolicy{}, handler.EnqueueRequestsFromMapFunc(r.getJobsToReconcile)).
+		Complete(r)
+}
+
+func (r *SKIPJobReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	skipJob := &skiperatorv1alpha1.SKIPJob{}
+	err := r.GetClient().Get(ctx, req.NamespacedName, skipJob)
+
+	if errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		r.EmitWarningEvent(skipJob, "ReconcileStartFail", "something went wrong fetching the SKIPJob, it might have been deleted")
+		return reconcile.Result{}, err
+	}
+
+	r.EmitNormalEvent(skipJob, "ReconcileStart", fmt.Sprintf("SKIPJob %v has started reconciliation loop", skipJob.Name))
+
+	err = skipJob.ApplyDefaults()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	controllerDuties := []func(context.Context, *skiperatorv1alpha1.SKIPJob) (reconcile.Result, error){
```

Comment:
`controllerDuties` is a good name for this :+1:

Reply:
Kudos to @evenh for the original function array in `application/controller.go` :)